app-backend/src/user/ServiceManager.js

```diff
@@ -452,12 +452,25 @@ class ServiceManager {
                     && (customDomain.length < 80)
                     && /^[a-z0-9\-\.]+$/.test(customDomain)
                     && (customDomain.indexOf('..') < 0)
-                    && (customDomain.indexOf(rootDomain) < 0 || (customDomain === rootDomain) || ((customDomain.indexOf(rootDomain) + rootDomain.length) !== customDomain.length));
+                    && (customDomain.indexOf("."+rootDomain) < 0 || (customDomain === rootDomain) || ((customDomain.indexOf(rootDomain) + rootDomain.length) !== customDomain.length));
 
-                if (!isAllowed) {
```

Comment:
`isAllowed` is not being used anymore - it's split into different conditions, but this condition is missed: `/^[a-z0-9\-\.]+$/.test(customDomain)`

Reply:
Once the condition is added, I think we can remove `isAllowed` as it will be an unused variable then.